File: java/src/main/java/dev/spiralmoon/maplestory/api/dto/CharacterSkillDTO.java
package dev.spiralmoon.maplestory.api.dto;

import com.google.gson.annotations.SerializedName;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.ToString;

import java.util.List;

/**
 * 캐릭터 스킬 정보
 */
@AllArgsConstructor
@Data
@ToString
public class CharacterSkillDTO {

    /**
     * 조회 기준일 (KST)
     */
    @SerializedName("date")
    private String date;

    /**
     * 캐릭터 직업
     */
    @SerializedName("character_class")
    private String characterClass;

    /**
     * 스킬 전직 차수
     */
    @SerializedName("character_skill_grade")
    private String characterSkillGrade;

    /**
     * 스킬 정보 리스트
     */
    @SerializedName("character_skill")
    private List<CharacterLinkSkillInfoDTO> characterSkill;
}

File: js/src/maplestory/api/dto/characterSkillDto.ts
import {CharacterSkillInfoDto} from "./characterSkillInfoDto";
import {CharacterSkillDtoBody} from "../response/characterSkillDtoBody";

/**
 * 캐릭터 스킬 정보
 */
class CharacterSkillDto {

    /**
     * 조회 기준일
     */
    date: Date;

    /**
     * 캐릭터 직업
     */
    characterClass: string;

    /**
     * 스킬 전직 차수
     */
    characterSkillGrade: string;

    /**
     * 스킬 정보
     */
    characterSkill: CharacterSkillInfoDto[];

    constructor(obj: CharacterSkillDtoBody) {
        const {
            date,
            character_class,
            character_skill_grade,
            character_skill,
        } = obj;

        this.date = new Date(date);
        this.characterClass = character_class;
        this.characterSkillGrade = character_skill_grade;
        this.characterSkill = character_skill.map(skill => new CharacterSkillInfoDto(skill));
    }
}

export {CharacterSkillDto};
